File: 0383.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        ransomNote = [str(i) for i in ransomNote]
        magazine = [str(i) for i in magazine]
        
        set_ransomNote = list(set(ransomNote))
        
        for i in set_ransomNote:
            a = ransomNote.count(i)
            b = magazine.count(i)
            
            if a > b:
                return False
                
        return True

File: cpp/0383.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        vector<int> c(26, 0);

        if (ransomNote.length() > magazine.length()) {
            return false;
        }

        for (int i = 0; i < ransomNote.length(); i++) {
            c[ransomNote[i] - 'a']++;
        }
        for (int i = 0; i < magazine.length(); i++) {
            c[magazine[i] - 'a']--;
        }

        for (int i = 0; i < 26; i++) {
            if (c[i] > 0) {
                return false;
            }
        }

        return true;
    }
};
